src/mpid/ch4/src/ch4r_win.h

```diff
@@ -25,6 +25,36 @@
         If false, performance-efficient mode is on, all allocated target
         objects are cached and freed at win_finalize.
 
+    - name        : MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL
+      category    : CH4
+      type        : int
+      default     : 1
+      class       : none
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_LOCAL
+      description : >-
+        Specifies the interval for polling progress of receive active message (AM)
+        in RMA communication. Effective only for passive target synchronization
+        MPI_Win_flush{_all} and MPI_Win_flush_local{_all}. Interval indicates the
+        number of performed flush calls before polling. It is counted globally
+        across all windows.
+        A value of 1 means always enable receive AM progress polling in every flush call.
+        A value of 0 means completely disable explicit progress polling for receive AM.
+        A value of -1 means the interval is automatically optimized at runtime.
```

Comment:
Can we make it an enum or boolean?

Reply:
The value may be an arbitrarily large integer as the interval. I am afraid mixing types will cause confusion.

Reply:
I think we probably want to rename this CVAR so it is clear as an opt-in switch. We are essentially removing the "high frequency" control as we discussed, right? I think trying to mixing the semantics of that control to the of/off switch control is confusing.

Reply:
Just for understanding, what is the difference between value `1` and `0`?

Reply:
> Just for understanding, what is the difference between value `1` and `0`?
> 
> It seems 0 means never poll AM progress in `MPI_Win_flush`, right? I don't think we need that option.

It is for developers/users who want to test with AM progress completely disabled. I think CVAR can be used for this purpose.

Reply:
Currently the model in my mind needs following two CVARs:
`MPIR_CVAR_CH4_RMA_ENABLE_SKIPPING_AM_PROGRESS`, and
`MPIR_CVAR_CH4_RMA_DEFAULT_AM_PROGRESS_INTERVAL`.

Reply:
> I think we probably want to rename this CVAR so it is clear as an opt-in switch. We are essentially removing the "high frequency" control as we discussed, right? I think trying to mix the semantics of that control to the on/off switch control is confusing.

Do you have any better CVAR definition in mind?

Reply:
Having a separate ENABLE CVAR is fine to me. But your suggested CVARs do not allow me to statically set the interval no matter whether incoming AM has been received. This functionality is an valid option for many users.

Reply:
> But your suggested CVARs do not allow me to statically set the interval no matter whether incoming AM has been received. This functionality is an valid option for many users.

The case is we know there is AM atomic messages, and we don't want to poll AM progress always? Wouldn't that guarantee the extra latency?

I assume you are not describing the case when there is AM atomic messages from *other* window, since I thought that is the case you want simplify away for now.

Reply:
"statically set the interval no matter whether incoming AM has been received" allows the user to tune the interval no matter the existence of recv_am. OFI has similar feature too, to allow user to tune the interval for waking up the async thread. I don't expect an end user to use it. But it is an useful thing for researchers or advanced users who wants to tune async progress v.s. latency.

> I assume you are not describing the case when there is AM atomic messages from other window, since I thought that is the case you want simplify away for now.
It is irrelevant to the above case.